app/controllers/UploadController.scala

```diff
@@ -19,6 +19,10 @@ class UploadController @Inject()(val controllerComponents: SecurityComponents,
                                 (implicit val ec: ExecutionContext) extends ValidatedActions with I18nSupport {
 
   def uploadPage(consignmentId: UUID): Action[AnyContent] = transferAgreementExistsAction(consignmentId) { implicit request: Request[AnyContent] =>
-    Ok(views.html.upload())
+    Ok(views.html.upload(consignmentId))
+  }
+
+  def uploadSubmit(consignmentId: UUID): Action[AnyContent] = secureAction { implicit request: Request[AnyContent] =>
```

Comment:
I think I was wrong about this action on the call! I thought we could remove it because the JavaScript will handle the redirect, but I was misremembering what we did in the prototype. I thought we did a pure-JavaScript redirect, but actually we just submitted the form when the upload had finished: https://github.com/nationalarchives/tdr-prototype-mvc/blob/master/js-src/upload/index.ts#L54

We'll still be using JavaScript to handle the submit click (see [Tom's open PR](https://github.com/nationalarchives/tdr-transfer-frontend/pull/74/files#diff-c03c04aaddc8881dc3dcf93441a5eed9R25)), but we will eventually submit the form.

So ignore what I said about deleting this controller action! It should stay. 🙂